onadata/settings/test_environ.py

```diff
@@ -99,18 +96,21 @@
 
 # MongoDB - moved here from common.py
 if MONGO_DATABASE.get('USER') and MONGO_DATABASE.get('PASSWORD'):
-    MONGO_CONNECTION_URL = (
-        "mongodb://%(USER)s:%(PASSWORD)s@%(HOST)s:%(PORT)s") % MONGO_DATABASE
+    MONGO_CONNECTION_URL = "mongodb://{user}:{password}@{host}:{port}/{db_name}".\
+        format(
+            user=MONGO_DATABASE['USER'],
+            password=quote_plus(MONGO_DATABASE['PASSWORD']),
+            host=MONGO_DATABASE['HOST'],
+            port=MONGO_DATABASE['PORT'],
+            db_name=MONGO_DATABASE['NAME']
+        )
 else:
-    MONGO_CONNECTION_URL = "mongodb://%(HOST)s:%(PORT)s" % MONGO_DATABASE
+    MONGO_CONNECTION_URL = "mongodb://%(HOST)s:%(PORT)s/%(NAME)s" % MONGO_DATABASE
 
-MONGO_CONNECTION = MongoClient(
-    MONGO_CONNECTION_URL, safe=True, j=True, tz_aware=True)
+MONGO_CONNECTION = MockMongoClient(
+    MONGO_CONNECTION_URL, j=True, tz_aware=True)
 MONGO_DB = MONGO_CONNECTION[MONGO_DATABASE['NAME']]
 
-# Clear out the test database
-if TESTING_MODE:
-    MONGO_DB.instances.drop()
```

Comment:
honestly, i would be happier if we stopped here `if 'instances' in settings.MONGO_DB.collection_names()` and required an acknowledgement before proceeding, maybe like the way Django reacts if its database already exists:
```
$ python manage.py test
Creating test database for alias 'default'...
Got an error creating the test database: database "myproject_test" already exists

Type 'yes' if you would like to try deleting the test database 'myproject_test', or 'no' to cancel:
```

that way, hopefully, we'd avoid modifying (and later destroying) any real data